From e638e41d085dccdcacc4393f842127f75c94f43c Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Thu, 7 Sep 2017 20:37:19 +0800
Subject: [PATCH 3317/5242] MLK-17491-22 clk: imx: clk-pllv4: add the missing
 CLK_SET_RATE_GATE

commit  53a904c2b4b0be221648fb3b6a61a51fa34fad2b from
https://source.codeaurora.org/external/imx/linux-imx.git

According to reference manual, this pll can't set rate without gating
clock. So we should add CLK_SET_RATE_GATE flag accordingly.

Cc: Anson Huang <Anson.Huang@nxp.com>
Cc: Bai Ping <ping.bai@nxp.com>
Fixes: 78ef764871d6 ("MLK-13441-5 ARM: imx: add new clk types")
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-pllv4.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/imx/clk-pllv4.c b/drivers/clk/imx/clk-pllv4.c
index 4f4c02b..27f4280 100644
--- a/drivers/clk/imx/clk-pllv4.c
+++ b/drivers/clk/imx/clk-pllv4.c
@@ -167,6 +167,7 @@ struct clk *imx_clk_pllv4(const char *name, const char *parent_name,
 	init.ops = &clk_pllv4_ops;
 	init.parent_names = &parent_name;
 	init.num_parents = 1;
+	init.flags = CLK_SET_RATE_GATE;
 
 	pll->hw.init = &init;
 
-- 
1.7.9.5

